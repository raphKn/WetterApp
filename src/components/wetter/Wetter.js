import React from 'react';
import PropTypes from 'prop-types';
import CustomSection from '../common/CustomSection';
import WetterEntryContainer from './WetterEntryContainer';

function renderWetterEntries(wetter) {
    return wetter.map(wetterentry => <WetterEntryContainer key={wetterentry.description} wetter={wetterentry} />);
}

const Wetter = (props) => {
    if (props.wetter.length < 1) {
        return null;
    }
    return (
        <CustomSection name="Beschreibungen">
            {renderWetterEntries(props.wetter)}
        </CustomSection>
    );
};

Wetter.propTypes = {
    wetter: PropTypes.arrayOf(
        PropTypes.shape({
            icon: PropTypes.string.isRequired,
            description: PropTypes.string.isRequired,
            main: PropTypes.string.isRequired,
        }),
    ).isRequired,
};

Wetter.defaultProps = {};

export default Wetter;
