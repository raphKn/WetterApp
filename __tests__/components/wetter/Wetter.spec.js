import React from 'react';
import renderer from 'react-test-renderer';
import Wetter from '../../../src/components/wetter/Wetter';

const regenwetter = {
    icon: 'RegenIcon',
    description: 'Nieselregen',
    main: 'Regen',
};
const sonnewetter = {
    icon: 'SonnenIcon',
    description: 'strahlender Sonnenschein',
    main: 'Sonne',
};
jest.mock('../../../src/components/wetter/WetterEntryContainer');

describe('Wetter', () => {
    it('renders two wetter correctly in a snapshot', () => {
        const wrapper = renderer.create(
            <Wetter wetter={[regenwetter, sonnewetter]} />);
        expect(wrapper).toMatchSnapshot();
    });
    it('renders nothing for no wetter in a snapshot', () => {
        const wrapper = renderer.create(
            <Wetter wetter={[]} />);
        expect(wrapper).toMatchSnapshot();
    });
});
