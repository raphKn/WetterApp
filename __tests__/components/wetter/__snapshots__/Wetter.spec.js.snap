// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wetter renders nothing for no wetter in a snapshot 1`] = `null`;

exports[`Wetter renders two wetter correctly in a snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "paddingTop": 12,
      },
      Object {
        "backgroundColor": "#3770a5",
      },
    ]
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#FAFAFA",
          },
          Object {
            "fontSize": 18,
            "paddingHorizontal": 18,
          },
        ]
      }
    >
      Beschreibungen
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#E0E0E0",
          "paddingHorizontal": 18,
          "paddingVertical": 6,
        },
        null,
      ]
    }
  />
</View>
`;
